### ISSUE TITLE
[BUG] `python-telegram-bot-raw` deprecation warning appears incorrectly if compiled with Nuitka

### ISSUE THREAD

**Issue by laggykiller** [python-telegram-bot/python-telegram-bot#4296]
### Steps to Reproduce

1. `pip install python-telegram-bot nuitka`
2. Create minimal python file `main.py` (The actual content does not matter as long as `telegram` is imported and used)
```python
from telegram.ext import ApplicationBuilder

app = ApplicationBuilder()
```
3. Run `main.py`. Note that deprecation warning is not appearing (Which is correct behaviour)
3. Compile with nuitka by `python -m nuitka --standalone main.py`
4. Run the compiled program. Note that deprecation warning is appearing incorrectly, even though `python-telegram-bot` is installed instead of `python-telegram-bot-raw`

```
PTBUserWarning: Hey. You seem to be using the `python-telegram-bot-raw` library. Please note that this libray has been deprecated and will no longer be updated. Please instead use the `python-telegram-bot` library. The change requires no changes in your code and requires no additional dependencies. For additional information, please see the channel post at https://t.me/pythontelegrambotchannel/145
```

### Expected behaviour

Deprecation warning should not appear when compiled with Nuitka

### Actual behaviour

Deprecation warning appears when compiled with Nuitka even if `python-telegram-bot` is used instead of `python-telegram-bot-raw`

### Operating System

Any (Recommend to test on Linux)

### Version of Python, python-telegram-bot & dependencies

```shell
python-telegram-bot 21.3
Bot API 7.4
Python 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)]
```

### Additional Context

Related issue: https://github.com/python-telegram-bot/python-telegram-bot/issues/4261

Related PR: https://github.com/python-telegram-bot/python-telegram-bot/pull/4270

Related code: https://github.com/python-telegram-bot/python-telegram-bot/blob/a9f6afd0152e30f32db69d35211fc05eceb9edfe/telegram/__init__.py#L482-L504

This is happening because `main.dist/telegram/ext` is missing in the Nuitka compilation result.

Instead of checking by `if not (Path(__file__).parent.resolve().absolute() / "ext").exists():`, I propose using this check instead:

```python
import importlib.utils

if importlib.util.find_spec("telegram.ext") is None:
    ...
```

**Comment by Poolitzer** [comment]
Did you verify it fixes the issue and still gives an error when you use ptb-raw?

**Comment by Bibo-Joshi** [comment]
Hi. Thanks for reaching out.

TBH I'm a bit confused by your description: In step one you say that you install ptb-raw. In step 5 you claim that you installed PTB instead of PTB-raw. Is step 1 a typo?

I'm not sure how common this use case is. I'm not familiar with nuitka and how it does the compilation. E.g. I'm wondering how nuitka can detect telegram.ext if it doesn't include that module in compilation...

Additionally, I'd like to point out that instead of doing a release that addresses the false positive, we could instead just get #4288 going and drop the warning from PTB completely:)

**Comment by laggykiller** [comment]
@Bibo-Joshi Yes that was a typo, corrected.

@Poolitzer The fix is working only if `main.py` (The script being compiled) uses `telegram.ext`. A workaround that I can confirm working is creating `main.dist/telegram/ext` directory manually after compiling. Another way would be checking `__compiled__ in globals()`, but this is overkill.

> I'm not sure how common this use case is.
Might also affect pyinstaller (?)

> we could instead just get 4288 going and drop the warning from PTB completely
Nice! No headache at all if that get merged.

**Comment by harshil21** [comment]
Closed by #4288
